apps/extension/src/ui/domains/SendFunds/SendFundsAccountPicker.tsx

```diff
@@ -29,6 +30,10 @@ export const SendFundsAccountPicker = () => {
         .filter((account) => {
           if (!token) return false
 
+          if (!chain?.hasCheckMetadataHash && account.ledgerApp) {
```

Comment:
This check isn't quite right. `account.ledgerApp` can be `substrate-legacy` or `substrate-generic` (see `SubstrateLedgerAppType`. If the chain does not have `hasCheckMetadataHash === true`, then we should allow accounts with `ledgerApp === 'substrate-legacy', as long as the genesis hash of the account matches the genesis hash of the chain.